tests/Equinox.MessageDb.Integration/AdjacentProjection.fs

```diff
@@ -0,0 +1,63 @@
+module Equinox.MessageDb.Integration.AdjacentProjection
+
+open Domain
+open Equinox.MessageDb
+open Dapper
+open Xunit
+open Swensen.Unquote
+
+let defaultBatchSize = 500
+let connectionString = "Host=localhost; Username=postgres; Password=; Database=message_store; Port=5432; Maximum Pool Size=10; Search Path=message_store, public"
+let connectToLocalStore _ =
+  MessageDbClient(connectionString)
+type Context = MessageDbContext
+type Category<'event, 'state, 'req> = MessageDbCategory<'event, 'state, 'req>
+
+let createContext connection batchSize = Context(connection, batchSize = batchSize)
+
+
+module Projection =
+    [<CLIMutable>]
+    type ContactPreferences = { id: string; many_promotions: bool; little_promotions: bool; product_review: bool; quick_surveys: bool }
+    let createTable (conn: Npgsql.NpgsqlConnection) =
+        conn.Execute("create table if not exists public.contact_preferences (
+            id text not null primary key,
+            many_promotions bool not null default false,
+            little_promotions bool not null default false,
+            product_review bool not null default false,
+            quick_surveys bool not null default false
+        )") |> ignore
+
+    let project streamId (state: ContactPreferences.Fold.State) (conn: Npgsql.NpgsqlConnection) = task {
+        let! _ = conn.ExecuteAsync(
```

Comment:
maybe add a note explaing why this is not upsert (are you trying to make it explode if erroneously called twice?)

Reply:
are you going to add the projection-pg upsert and delete helpers :D

Reply:
Upsert in place now